nixos/modules/services/networking/vpnc.nix

```diff
@@ -0,0 +1,97 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+
+  buildVpnc = key: value: {
+    name = "vpnc/${key}.conf";
+    value = {
+      text = ''
+        IPSec gateway ${value.gateway}
+        IPSec ID ${value.id}
+        IPSec secret ${value.secret}
+        Xauth username ${value.username}
+        Password helper ${pkgs.systemd}/bin/systemd-ask-password
+        Script ${pkgs.vpnc}/etc/vpnc/vpnc-script
+        '';
+    };
+  };
+  up = pkgs.writeScript "up" ''
+    #!${pkgs.bash}/bin/bash
+    PATH=${makeBinPath [pkgs.nettools pkgs.iproute]}:$PATH
+
+    LOCAL_GATEWAY=$(ip r | grep "default via" | grep -oE "([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)" | head -n1)
+    ip route del default via $LOCAL_GATEWAY
+    ip route add default dev tun0 proto static scope link metric 50
+    if [ ! -d /var/run/vpnc ]; then
```

Comment:
`/var/run` is deprecated. Also, use `RuntimeDirectory=vpnc` to create that directory